### PR TITLE
Code for dynamic & relative email campaigns and verifications

### PR DESCRIPTION
**Pull request by simonorchanian**

### CODE REVIEW

**Comment by dakaugu** [comment]
squash commits  into 1 commit

File: Code/src/PageObjects/AdminEmailRelativeCampaigns.java [hunk 1]
```diff
@@ -176,9 +178,16 @@ public boolean verifyRelativeEmailCampaignOnUser() throws InterruptedException {
...
 			 Thread.sleep(2000);
-			 return true; 
+			 
+			 WebElement getCampaignName = driver.findElement(By.cssSelector("tr:nth-child(2) > td:nth-child(1) > div > a"));
+			 String userCampaignName = getCampaignName.getText();
+			 if (userCampaignName.contains("Hearsay Systems Automation"))
+					 {
+			 
+			 return true;  
 			}
```
Comment (dakaugu) [inline Code/src/PageObjects/AdminEmailRelativeCampaigns.java]: can we reformat this whole block. indentation is not correct

File: Code/src/Tests/AdminEmailDynamicCampaignsTest.java [hunk 2]
```diff
@@ -54,22 +53,13 @@ public void verifyAdminEmailCampaigns(String username, String password) throws I
 			
 			System.out.println("Dynamic email campaigns: check the live status message");
 			Assert.assertEquals(adminEmailCampaigns.verifyLiveStatusOnCampaign(), "Live");
-			
-			//System.out.println("Dynamic email campaigns: verify dynamic campaign on user view"); 
-			//Assert.assertTrue(adminEmailCampaigns.verifyDynamicEmailCampaignOnUser());
+
```
Comment (dakaugu) [inline Code/src/Tests/AdminEmailDynamicCampaignsTest.java]: lets remove those empty blank lines too

File: Code/src/PageObjects/AdminEmailRelativeCampaigns.java [hunk 3]
```diff
@@ -127,7 +129,7 @@ public String composeEmailFormFillout() throws InterruptedException {
 			    if(composeEmailScheduleLink.isDisplayed()) 
 			    {
 				  composeEmailScheduleLink.click();
-				  Thread.sleep(2000);
+				  Thread.sleep(4000);
```
Comment (rogerhu) [inline Code/src/PageObjects/AdminEmailRelativeCampaigns.java]: Is there nothing in the Java Selenium that does implicit/explicit waits?

https://selenium-python.readthedocs.io/waits.html

File: Code/src/Tests/AdminEmailDynamicCampaignsTest.java (hunk 2 above)
Comment (simonorchanian) [reply Code/src/Tests/AdminEmailDynamicCampaignsTest.java]: Done

File: Code/src/PageObjects/AdminEmailRelativeCampaigns.java (hunk 1 above)
Comment (simonorchanian) [reply Code/src/PageObjects/AdminEmailRelativeCampaigns.java]: Done

File: Code/src/PageObjects/AdminEmailRelativeCampaigns.java (hunk 3 above)
Comment (simonorchanian) [reply Code/src/PageObjects/AdminEmailRelativeCampaigns.java]: There is and I've used it in other places for best practices ;  
However, I've found it doesn't work great so I'll place sleeps sometimes.

**Comment by dakaugu** [comment]
can we select everything on the file and run ⌘ + ⌥ + L if using jet brains IDE. there seems to still be indentation and format issues

**Comment by simonorchanian** [comment]
@dakaugu :  I'm using eclipse but I've corrected the indentation using their option.....

**Comment by dakaugu** [comment]
nice. let's remove some of those whitespaces now

File: Code/src/PageObjects/AdminEmailRelativeCampaigns.java [hunk 4]
```diff
@@ -2,200 +2,207 @@
...
-	
-	
-	
-	
\ No newline at end of file
+		return null;
+
+	}
+
+
```
Comment (dakaugu) [inline Code/src/PageObjects/AdminEmailRelativeCampaigns.java]: unnecessary whitespaces

File: Code/src/PageObjects/AdminEmailRelativeCampaigns.java [hunk 5]
```diff
@@ -2,200 +2,207 @@
...
+	@Override
+	public void method2() {
+		// TODO Auto-generated method stub
+
+	}
+
+}
+
+
```
Comment (dakaugu) [inline Code/src/PageObjects/AdminEmailRelativeCampaigns.java]: whitespaces

File: Code/src/Tests/AdminEmailRelativeCampaignsTest.java [hunk 6]
```diff
@@ -1,98 +1,83 @@
...
+}
+
+
+
+
+
+
+
+
+
```
Comment (dakaugu) [inline Code/src/Tests/AdminEmailRelativeCampaignsTest.java]: blank lines

**Comment by dakaugu** [comment]
great! this looks ready for a proper review now

**Comment by simonorchanian** [comment]
Thank you.

**Comment by simonorchanian** [comment]
@dakaugu:  Is this PR ready to be merged in ?  Thank you.

**Pull request merged**